event/handler.go

```diff
@@ -239,3 +248,28 @@ func (h *ImageUploadedHandler) createVaultKeyForFile(keyPath string) ([]byte, er
 
 	return psk, nil
 }
+
+func (h *ImageUploadedHandler) setImageStatusToFailed(ctx context.Context, imageID string, desc string) {
+	image, err := h.ImageCli.GetImage(ctx, "", h.AuthToken, "", imageID)
+	if err != nil {
+		log.Event(ctx, "error getting image from API to set failed_import status", log.ERROR, log.Error(err))
+		return
+	}
+	image.State = failedState
+	image.Error = desc
+	_, err = h.ImageCli.PutImage(ctx, "", h.AuthToken, "", imageID, image)
```

Comment:
Consider using a `PATCH` request for this update. This would prevent you having to use a `GET` request before the `PUT`

Reply:
I think this is a very good point, but requires some more thinking on how this is implemented with the use of IF-Match headers and ETags. I will create a story in the backlog for triage for the tech leads to discuss but this shouldn't hold up this pull request. Thanks Carl